### ISSUE TITLE
Rust clippy throws compiler error when trying to fix an uneeded loop variable 

### ISSUE THREAD

**Issue by MitchelPaulin** [rust-lang/rust-clippy#9396]
### Summary

Getting a weird error where rust is trying to remove the erroneous loop variable but gets caught on removing `mut`, likely because I don't use `i` to index into the array even though it seems like what I would be doing. 

I got this error in a larger project but managed to reproduce it with pretty minimal code below 

### Reproducer

I tried this code:

```rust
fn main() {
    let mut i: usize = 0;
    let test_list: Vec<usize> = vec![1,2,3];
    for x in test_list {
        println!("{} {}", i, x);
        i += 1;
    }
}
```

I expected to see this happen:
It would not change the code since I use `i` in the print statement 

Instead, this happened:
It tried to simplify the code and threw an error

### Output 

```bash
    Checking test-for-clippy v0.1.0 (/home/mitch/git-projects/test-for-clippy)
warning: failed to automatically apply fixes suggested by rustc to crate `test_for_clippy`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see 
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
warning: unused variable: `i`
 --> src/main.rs:2:13
  |
2 |     let mut i: usize = 0;
  |             ^ help: if this is intentional, prefix it with an underscore: `_i`
  |
  = note: `#[warn(unused_variables)]` on by default

warning: value assigned to `i` is never read
 --> src/main.rs:6:9
  |
6 |         i += 1;
  |         ^
  |
  = note: `#[warn(unused_assignments)]` on by default
  = help: maybe it is overwritten before being read?

warning: variable does not need to be mutable
 --> src/main.rs:2:9
  |
2 |     let mut i: usize = 0;
  |         ----^
  |         |
  |         help: remove this `mut`
  |
  = note: `#[warn(unused_mut)]` on by default

error[E0384]: cannot assign twice to immutable variable `i`
 --> src/main.rs:6:9
  |
4 |     for (i, x) in test_list.into_iter().enumerate() {
  |          -
  |          |
  |          first assignment to `i`
  |          help: consider making this binding mutable: `mut i`
5 |         println!("{} {}", i, x);
6 |         i += 1;
  |         ^^^^^^ cannot assign twice to immutable variable

error: aborting due to previous error; 3 warnings emitted

For more information about this error, try `rustc --explain E0384`.
Original diagnostics will follow.

warning: the variable `i` is used as a loop counter
 --> src/main.rs:4:5
  |
4 |     for x in test_list {
  |     ^^^^^^^^^^^^^^^^^^ help: consider using: `for (i, x) in test_list.into_iter().enumerate()`
  |
  = note: `#[warn(clippy::explicit_counter_loop)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#explicit_counter_loop

warning: `test-for-clippy` (bin "test-for-clippy") generated 1 warning
warning: `test-for-clippy` (bin "test-for-clippy" test) generated 1 warning (1 duplicate)
    Finished dev [unoptimized + debuginfo] target(s) in 0.07s 
```

### Version

```text
rustc 1.63.0 (4b91a6ea7 2022-08-08)
binary: rustc
commit-hash: 4b91a6ea7258a947e59c6522cd5898e7c0a6a88f
commit-date: 2022-08-08
host: x86_64-unknown-linux-gnu
release: 1.63.0
LLVM version: 14.0.5
```

**Comment by kraktus** [comment]
Hey, thanks for reporting!

`explicit_counter_loop` applicability has been switched to `MaybeIncorrect` since, meaning the suggestion need to be manually applied by the programmer. Considering the suggestion is correct I'd say this issue is fixed 👍
